src/sparseml/optim/manager.py

```diff
@@ -93,6 +103,7 @@ def compose_staged(
         cls,
         base_recipe: Union[str, "BaseManager"],
         additional_recipe: Union[str, "BaseManager"],
+        additional_metadata=Dict[str, Any],
```

Comment:
as discussed in call, let's get rid of additional metadata. this should be completely controlled by the given recipe/manager objects.

every stage on either side of the merge should keep its original metadata in the final composed recipe